orchard_key_components.py

```diff
@@ -37,6 +38,14 @@ def diversify_hash(d):
         P = group_hash(b'z.cash:Orchard-gd', b'')
     return P
 
+def prf_nf_orchard(nk, rho):
+    return poseidon_hash(nk, rho)
+
+def derive_nullifier(nk, rho: Fp, psi: Fp, cm):
+    scalar = to_base(prf_nf_orchard(nk, rho)) + psi  # addition mod p
```

Comment:
`poseidon_hash` already returns `Fp`, I think, in which case the `to_base` shouldn't be necessary.